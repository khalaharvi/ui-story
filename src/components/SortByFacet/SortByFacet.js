import React from 'react';
import PropTypes from 'prop-types';
import { connectSortBy } from 'react-instantsearch/connectors';
import { cx as emotionCx } from 'react-emotion';

import { createClassNames } from 'utils';
import Select from 'components/Select';

export const cx = createClassNames('sortby-facet');

export const SortByFacet = ({
  className,
  items,
  currentRefinement,
  refine
}) => (
  <Select
    className={emotionCx(cx(''), className)}
    items={items}
    value={currentRefinement}
    onChange={value => refine(value)}
  />
);

SortByFacet.propTypes = {
  className: PropTypes.string,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      value: PropTypes.string.isRequired
    })
  ).isRequired,
  currentRefinement: PropTypes.string.isRequired,
  refine: PropTypes.func.isRequired
};

export default connectSortBy(SortByFacet);
