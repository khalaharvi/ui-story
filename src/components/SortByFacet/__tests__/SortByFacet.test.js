import React from 'react';
import { mount, shallow } from 'enzyme';

import { SortByFacet } from '../SortByFacet';

describe('<SortByFacet />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SortByFacet
        refine={() => null}
        items={[
          { value: 'index1', label: 'index name 1' },
          { value: 'index2', label: 'index name 2' },
          { value: 'index3', label: 'index name 3' },
          { value: 'index4', label: 'index name 4' }
        ]}
        currentRefinement="index1"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with className', () => {
    const wrapper = shallow(
      <SortByFacet
        refine={() => null}
        items={[
          { value: 'index1', label: 'index name 1' },
          { value: 'index2', label: 'index name 2' },
          { value: 'index3', label: 'index name 3' },
          { value: 'index4', label: 'index name 4' }
        ]}
        currentRefinement="index1"
        className="custom-sortby-facet"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('refines its value on change', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <SortByFacet
        refine={refine}
        items={[
          { value: 'index1', label: 'index name 1' },
          { value: 'index2', label: 'index name 2' },
          { value: 'index3', label: 'index name 3' },
          { value: 'index4', label: 'index name 4' }
        ]}
        currentRefinement="index1"
      />
    );

    const selectedValue = wrapper.find('select');
    expect(selectedValue.find('option')).toHaveLength(4);
    expect(
      selectedValue
        .find('option')
        .first()
        .text()
    ).toBe('index name 1');

    selectedValue
      .find('select')
      .simulate('change', { target: { value: 'index3' } });

    expect(refine.mock.calls).toHaveLength(1);
    expect(refine.mock.calls[0][0]).toEqual('index3');
  });

  it('should use value if no label provided', () => {
    const wrapper = mount(
      <SortByFacet
        refine={() => null}
        items={[
          { value: 'index1' },
          { value: 'index2' },
          { value: 'index3' },
          { value: 'index4' }
        ]}
        currentRefinement="index1"
      />
    );

    const selectedValue = wrapper.find('select');
    expect(selectedValue.find('option')).toHaveLength(4);
    expect(
      selectedValue
        .find('option')
        .first()
        .text()
    ).toBe('index1');
  });
});
