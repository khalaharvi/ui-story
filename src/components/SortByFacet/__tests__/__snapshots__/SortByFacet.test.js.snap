// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortByFacet /> renders correctly 1`] = `
<Select
  className="dh-sortby-facet"
  items={
    Array [
      Object {
        "label": "index name 1",
        "value": "index1",
      },
      Object {
        "label": "index name 2",
        "value": "index2",
      },
      Object {
        "label": "index name 3",
        "value": "index3",
      },
      Object {
        "label": "index name 4",
        "value": "index4",
      },
    ]
  }
  onChange={[Function]}
  value="index1"
/>
`;

exports[`<SortByFacet /> renders correctly with className 1`] = `
<Select
  className="dh-sortby-facet custom-sortby-facet"
  items={
    Array [
      Object {
        "label": "index name 1",
        "value": "index1",
      },
      Object {
        "label": "index name 2",
        "value": "index2",
      },
      Object {
        "label": "index name 3",
        "value": "index3",
      },
      Object {
        "label": "index name 4",
        "value": "index4",
      },
    ]
  }
  onChange={[Function]}
  value="index1"
/>
`;
